### PR TITLE
Orders with multiple random qualities

### PR DESCRIPTION
**Pull request by CarstenWalther**
Previously, a merchant ordered a specific amount of a random product with a random quality. With this PR, merchants order a specific amount and get items of one random product type but of different qualities. The qualities are chosen uniformly randomly from the available qualities.

Unit price, fixed order costs and product stock is now unique for each product type. Previously these attributes were unique for each product type and quality combination. This change has the advantage, that merchants can calculate how much an order will cost them.

The producer API changed. E.g. an order contains a list of products instead of one product. Existing merchants won't work with these changes. I'll create a pull request with updated example merchants. There will also be a change in the analytics repository.

Currently it's not possible to inspect or change the producer configuration in the management UI. I'll need to change the management UI as well to work with the new product information format.

If a merchant orders a product, he will receive an order with the following attributes:

- product id
- product name
- billing amount
- fixed cost
- unit price
- stock
- left in stock
- list of products

And products in that order have the following attributes:
- uid
- product id
- name
- quality
- amount
- time to life
- start of lifetime
- signature

### CODE REVIEW

**Comment by CarstenWalther** [comment]
This pull request is work in progress. Please do not merge yet

**Comment by CarstenWalther** [comment]
I updated the communication between producer and management UI and removed some API routes that are not used. For now, I am done with this pull request.

File: controllers/models/Products.js [hunk 1]
```diff
@@ -1,267 +1,199 @@
...
+
+const allQualities = Object.freeze({veryGood: 1, good: 2, fair: 3, acceptable: 4});
+
+const Products = {
+
+    // This is a mapping from product id to product info
+    productsInfo: {
+        1: {
+            product_id: 1,
+            name: 'CD_1',
```
Comment (Bouncner) [inline controllers/models/Products.js]: Can we rename to `Product #1`?

File: controllers/models/Products.js [hunk 2]
```diff
@@ -1,267 +1,199 @@
...
+const aesEcb = new aesjs.ModeOfOperation.ecb(secret_key);
+
+const allQualities = Object.freeze({veryGood: 1, good: 2, fair: 3, acceptable: 4});
+
+const Products = {
+
+    // This is a mapping from product id to product info
+    productsInfo: {
+        1: {
+            product_id: 1,
```
Comment (Bouncner) [inline controllers/models/Products.js]: Can we find a solution that does not require the repetition of the ID?

**Review by Bouncner — CHANGES_REQUESTED** [review]

**Review by Bouncner — APPROVED** [review]

**Pull request merged**